storage/database.go

```diff
@@ -104,16 +107,23 @@ type databaseMetrics struct {
 	unknownNamespaceWriteTagged         tally.Counter
 	unknownNamespaceFetchBlocks         tally.Counter
 	unknownNamespaceFetchBlocksMetadata tally.Counter
+	unknownNamespaceQueryIDs            tally.Counter
+	queryIDsIndexDisabled               tally.Counter
+	writeTaggedIndexDisabled            tally.Counter
 }
 
 func newDatabaseMetrics(scope tally.Scope) databaseMetrics {
 	unknownNamespaceScope := scope.SubScope("unknown-namespace")
+	indexDisabledScope := scope.SubScope("index-disabled")
 	return databaseMetrics{
 		unknownNamespaceRead:                unknownNamespaceScope.Counter("read"),
 		unknownNamespaceWrite:               unknownNamespaceScope.Counter("write"),
 		unknownNamespaceWriteTagged:         unknownNamespaceScope.Counter("write-tagged"),
 		unknownNamespaceFetchBlocks:         unknownNamespaceScope.Counter("fetch-blocks"),
 		unknownNamespaceFetchBlocksMetadata: unknownNamespaceScope.Counter("fetch-blocks-metadata"),
+		unknownNamespaceQueryIDs:            unknownNamespaceScope.Counter("query-ids"),
+		queryIDsIndexDisabled:               indexDisabledScope.Counter("query-ids"),
+		writeTaggedIndexDisabled:            indexDisabledScope.Counter("write-tagged"),
```

Comment:
nit: rename these to be explicit about them being error cases?  `errQueryIDsIndexDisabled`, etc?  And for the names perhaps "err-query-ids" and "err-write-tagged"?